### PR TITLE
Feat -- Namespaces models

### PR DESCRIPTION
**Pull request by bryonbean**
- autoloader now lazy loads models with, e.g. `use CDash\Model\Build` vs require
- a more conventional MVC directory structure

### CODE REVIEW

File: include/CDash/Database.php [hunk 1]
```diff
@@ -15,7 +15,6 @@
 =========================================================================*/
 namespace CDash;
 
-use CDash\Log;
```
Comment (zackgalbreath) [inline include/CDash/Database.php]: this turned out to be unnecessary?  Are we not using Log inside Database?

File: include/autoremove.php [hunk 2]
```diff
@@ -108,7 +111,6 @@ function removeFirstBuilds($projectid, $days, $maxbuilds, $force = false)
     // Remove any job schedules that are older than our cutoff date
     // and not due to repeat again.
     require_once 'models/constants.php';
```
Comment (zackgalbreath) [inline include/autoremove.php]: this can probably go away too, right?

File: include/clientsubmit.php [hunk 3]
```diff
@@ -14,21 +14,22 @@
...
+use CDash\Model\ClientOS;
+use CDash\Model\ClientJobSchedule;
+use CDash\Model\ClientCompiler;
+use CDash\Model\ClientCMake;
+use CDash\Model\ClientLibrary;
+use CDash\Model\ClientJob;
+
 function client_submit()
 {
-    include 'config/config.php';
```
Comment (zackgalbreath) [inline include/clientsubmit.php]: ??? this diff looks funny to me

File: include/common.php [hunk 4]
```diff
@@ -87,7 +92,7 @@ function generate_XSLT($xml, $pageName, $only_in_local = false)
     // For common xsl pages not referenced directly
     // i.e. header, headerback, etc...
     // look if they are in the local directory, and set
-    // an XML value accordingly
+    // an XML value accordinglyClientLibrary
```
Comment (zackgalbreath) [inline include/common.php]: bad copy/paste perhaps?

File: include/CDash/Database.php (hunk 1 above)
Comment (bryonbean) [reply include/CDash/Database.php]: This is still using `add_log` so, currently, CDash\Log is unused.

File: include/autoremove.php (hunk 2 above)
Comment (bryonbean) [reply include/autoremove.php]: No, this should go away. Good catch.

File: include/clientsubmit.php (hunk 3 above)
Comment (bryonbean) [reply include/clientsubmit.php]: Yeah, that is funky, but I think it's correct.

**Pull request merged**